internal/utils/mocks.go

```diff
@@ -1,4 +1,4 @@
-package mocks
+package utils
```

Comment:
Renamed this package and file since I put the grpc interceptor stuff in here also.

Reply:
idk if we care about this, but it looks like the rest of this file's changes are just linting? Might want to remove (or at least include in this PR which linter you're using locally, so all contributors can use it so we don't go back and forth on linting changes in the future).

Reply:
I am using the Go plugin for vscode with the default linting settings. I'm using `staticcheck`, it looks like this: 

![Screenshot 2024-07-08 at 1 29 42 PM](https://github.com/pinecone-io/go-pinecone/assets/119623786/d9a55bf9-4799-40b3-99fd-ec19ceac3e06)

Tbh I'm not clear why it introduced a few linting changes here. I wrote the file originally and I don't think it has been changed since. I may not have had format-on-save enabled when I originally wrote it.